File: 30 Days of Code/Python/26 - Day 25 - Running Time and Complexity.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem
# Difficulty: Medium
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

# Note: 1 is NOT a prime number, that is, a number divisable by another number less or equal to the square root of the first number, it is NOT prime

for _ in range(int(input())):
    num = int(input())
    if num == 1:
        print("Not prime")
    else:
        if(num % 2 == 0 and num > 2):
            print("Not prime")
        else:
            for i in range(3, int(num**(1/2))+1, 2):
                if num % i == 0:
                    print("Not prime")
                    break
            else:
                print("Prime")

File: 30 Days of Code/C#/26 - Day 25 - Running Time and Complexity.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem
// Difficulty: Medium
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

// Note: 1 is NOT a prime number, that is, a number divisable by another number less or equal to the square root of the first number, it is NOT prime

using System;

class Solution {
	static void Main(String[] args) {
		int numberOfInputs = Convert.ToInt32(Console.ReadLine());
		int i, j, inputNumber;
		string output;

		for (i = 1; i <= numberOfInputs; i++) {
			inputNumber = Convert.ToInt32(Console.ReadLine());

			for (j = 2; j <= inputNumber / j; j++) {
				if (inputNumber % j == 0) {
                    inputNumber = 1;
                }
			}
			output = inputNumber == 1 ? "Not prime": "Prime";
			Console.WriteLine(output);
		}
	}
}
